app/models/api/hses.rb

```diff
@@ -18,8 +18,17 @@ def request
   end
 
   class Issues
+    def initialize(user:)
+      @username = user["uid"]
+    end
+
     def request
-      @issues ||= FakeIssues.instance.json
+      res = Api.query_api(
```

Comment:
[question] Why call back up to the `Api` level for this. We know we'll have multiple API endpoints so might as well keep that separation now.

Reply:
The behavior of sending and receiving a response felt like fairly generic behavior. I realize that the authentication methods may vary, but it still feels like there ought to be some central class that handles it. Then maybe we have a subclass for the different systems that override some of the implementation but continue to rely on the same error handling / etc.  I dunno, I don't really want to move it back because it felt very weird in the `Issues` class....maybe I can find a home in an Hses-specific class that would make me feel better.

Reply:
[comment/suggestion] i think it should be on the Api class -- but i think we should reorganize this a bit to have Api just hold the http behavior and the individual systems hold their own config information. what i am envisioning right now is this: 

if we have the classes inherit from Api then they will get the classes from them. so we can add 
`class Api::Hses < Api` (is it weird to namespace AND inherit from there? i say probably, so maybe we want to have one or both be different?) and then we could have thinks like 

```ruby
class Api::Hses < Api
  def host
    Rails.configuration.x.hses.api_base
  end
  def path
  end
  def query
  end
end
```

and then in Api we could have 
```ruby
class Api
...
  def query_api
    uri = build_uri(host, path, query)
    res = send_api_request(uri)
    body = format_response_body(res.body)
    {code: res.code, body: body}
  end
```

so this line could just be 
`res = query_api`

or we could even have `res` be its own method so:

```ruby
def request
  {code: res.code, body: format_response_body(res.body)}
end

def res
  @res ||= query_api
end
```

i like this because then we don't have to keep track in Api or somewhere what the configuration of each system is -- the system gets to keep track of it. and if there are different queries for specific endpoints they can override those by defining their own `query` method!

what do you all think?

Reply:
Totally agree.  I was thinking of something along the same lines.  Basically:

- ApiDelegator: has logic which helps you get to the correct type of Api subclass (Hses / FakeData::Hses / Tta, etc)
- Api:  has defaults like you're describing, including stuff like logic for pagination and sorting
- Api subclasses:  have the specific implementation details and overrides

That makes wayyy more sense to me.  Want me to implement that on this PR?

Reply:
oh yeah i like this structure! i would love to see it on this PR!